plugins/commands/watch.py

```diff
@@ -107,21 +110,60 @@ def docker(command):
     return subprocess.run(prefix + ["docker"] + command)
 
 
+class WatchingBuilder(object):
+
+    def __init__(self, incontext):
+        self.incontext = incontext
+
+    def start(self):
+        self.builder = Builder(self.incontext)
+        self.builder.start()
+        logging.info("Watching directory...")
+        self.observer = watch_directory([self.incontext.configuration.site.paths.content,
+                                         self.incontext.configuration.site.paths.templates],
+                                        self.builder.schedule)
+        logging.info("Performing initial build...")
+        self.builder.schedule()
+
+    def stop(self):
+        self.builder.stop()
+        self.observer.stop()
+        self.observer.join()
+        self.builder.join()
+
+
 @incontext.command("watch", help="watch for changes and automatically build the website")
 def command_watch(incontext, options):
-    builder = Builder(incontext)
+    builder = WatchingBuilder(incontext)
     builder.start()
-    logging.info("Watching directory...")
-    observer = watch_directory([incontext.configuration.site.paths.content,
-                                incontext.configuration.site.paths.templates],
-                               builder.schedule)
-    logging.info("Performing initial build...")
-    builder.schedule()
     try:
         while True:
             time.sleep(0.2)
     except KeyboardInterrupt:
         builder.stop()
-        observer.stop()
-    observer.join()
-    builder.join()
+
+
+@incontext.command("serve", help="run a local web server for development", arguments=[
```

Comment:
It would be great if this functionality were also in the new `cli` module as it would make for a more consistent API. If this changes, it's important to check whether there's any documentation which also needs updating.